src/core/row.cpp

```diff
@@ -109,3 +137,156 @@ std::size_t row::find_column(std::string const &name) const
 
     return it->second;
 }
+
+std::pair<std::string*, indicator*> row::alloc_data_holder_string()
+{
+    data_holder *holder = NULL;
+    std::string *s = new std::string();
```

Comment:
This should be replaced with
```suggestion
    cxx_details::auto_ptr<data_holder> holder(new data_holder());
    cxx_details::auto_ptr<std::string> s(new std::string());
```
see `include/soci/soci-platform.h` and then `holder.release` and `s.release()` should be used below to avoid managing memory manually.